test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,162 @@
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-3
+              pod: pod-1
+              phase: Failed
+            exp_annotations:
+              summary: "Pod pod-1 is not ready"
+              description: "Pod pod-1 in namespace ns-3 is not ready for more than 15 minutes."
+
+  - interval: 1m
+    input_series:
+      # Not Alerted cases:
+      # Pod is in the Unknown state and unscheduled (unscheduled once in the 15 min interval), so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Unknown"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-4", pod="pod-1"}'
+        values: '1x0 0x13'
+
+      # Pod is in Succeeded state and scheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Unknown"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Succeeded"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-5", pod="pod-1"}'
+        values: '0x14'
+
+      # For the first 5 minutes, Pod is in the Pending state, next 5 minutes it's in the Unknown state, for last 5 minutes it's in the Failed state and scheduled, so it will be not alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Pending"}'
+        values: '1x4 0x4 0x4'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Unknown"}'
+        values: '0x4 1x4 0x4'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Failed"}'
+        values: '0x4 0x4 1x4'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-6", pod="pod-1"}'
+        values: '0x14'
+
+      # Pod is in the Running state for the first 14 min, then it's in the Failed state for the last 1 min and unscheduled, so it will not be alerted.
```

Comment:
```suggestion
      # Pod is in the Running state for the first 14 min, then it's in the Failed state
      # for the last 1 min and unscheduled, so it will not be alerted.
```